scripts/west_commands/build.py

```diff
@@ -557,7 +562,34 @@ def _run_cmake(self, board, origin, cmake_opts):
         if user_args:
             cmake_opts.extend(shlex.split(user_args))
 
-        config_sysbuild = config_getboolean('sysbuild', False)
+        # Check if this is an ncs-repo directory
+        allow_list = ['zephyr', 'mcuboot', 'sidewalk', 'find-my', 'nrf']
+        in_ncs_repo_dir = False
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            if module.meta['name'] in allow_list:
+                if self.source_dir[0:len(module.project)] == module.project:
+                    path_char = self.source_dir[len(module.project):len(module.project)+1]
+
+                    if path_char == '/' or path_char == '\\':
```

Comment:
Just use the built-in pathlib features, like:
```
Path(source_dir).relative_to(module.project)
```

Reply:
Updated